timesketch/api/v1/resources/analysis.py

```diff
@@ -186,25 +221,29 @@ def post(self, sketch_id):
         # Import here to avoid circular imports.
         # pylint: disable=import-outside-toplevel
         from timesketch.lib import tasks
-        try:
-            analyzer_group, session_id = tasks.build_sketch_analysis_pipeline(
-                sketch_id=sketch_id, searchindex_id=search_index.id,
-                user_id=current_user.id, analyzer_names=analyzers,
-                analyzer_kwargs=analyzer_kwargs)
-        except KeyError as e:
-            return abort(
-                HTTP_STATUS_CODE_BAD_REQUEST,
-                'Unable to build analyzer pipeline, analyzer does not exist. '
-                'Error message: {0!s}'.format(e))
 
-        if analyzer_group:
-            pipeline = (tasks.run_sketch_init.s(
-                [search_index.index_name]) | analyzer_group)
-            pipeline.apply_async()
+        sessions = []
+        for timeline_id in timeline_ids:
+            timeline = Timeline.query.get(timeline_id)
+            searchindex_id = timeline.searchindex.id
+            searchindex_name = timeline.searchindex.index_name
+
+            try:
+                analyzer_grp, session = tasks.build_sketch_analysis_pipeline(
+                    sketch_id=sketch_id, searchindex_id=searchindex_id,
+                    user_id=current_user.id, analyzer_names=analyzers,
+                    analyzer_kwargs=analyzer_kwargs)
+            except KeyError as e:
+                return abort(
+                    HTTP_STATUS_CODE_BAD_REQUEST,
+                    'Unable to build analyzer pipeline, analyzer does not '
+                    'exist. Error message: {0!s}'.format(e))
 
-        schema = {
-            'objects': [{
-                'analysis_session': session_id
-            }]
-        }
-        return jsonify(schema)
+            if analyzer_grp:
+                pipeline = (tasks.run_sketch_init.s(
+                    [searchindex_name]) | analyzer_grp)
+                pipeline.apply_async()
+
+            sessions.append(session)
+
+        return self.to_json(sessions)
```

Comment:
This will also mess up the API client, which expects to get the other schema that was previously returned, so you'll have to fix the API client here